### PR TITLE
Add Fast parsing for kube cluster and namespace

### PR DESCRIPTION
**Pull request by spheromak**
Parses kube config directly (as forking/execing kubectl is super slow)  to extract kube cluster and namespace info. 

image

---
This change is [Reviewable](https://reviewable.io/reviews/justjanne/powerline-go/40)

### CODE REVIEW

**Comment by justjanne** [comment]
This is a great idea, but you’ve accidentally removed some line breaks in the Readme (markdown uses two spaces at the end of the line for <br>), could you add those back?

**Comment by spheromak** [comment]
sorry my editor cleans trailing whites pace. (markdown is ridiculous to use 2 spaces :\ )  I'll fix it up latter today.

File: segment-kube.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	namespace := ""
+	for _, c := range kc.Contexts {
+		if c.Name == kc.CurrentContext {
+			cluster = c.Context.Cluster
+			namespace = c.Context.Namespace
+			break
+		}
+	}
+
+	if strings.HasPrefix(cluster, "gke") {
```
Comment (joemiller) [inline segment-kube.go]: This should be commented to indicate we're truncating google container engine (GKE) cluster names from long-form to a shorter more human friendly form, eg:

```
gke_projectname_availability-zone_cluster-01 => cluster-01
```

File: segment-kube.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+type KubeConfig struct {
+	Contexts       []KubeContext `yaml:"contexts"`
+	CurrentContext string        `yaml:"current-context"`
+}
+
+func segmentKube(p *powerline) {
+	home, _ := os.LookupEnv("HOME")
+	defaultConfigFile := home + "/.kube/config"
```
Comment (joemiller) [inline segment-kube.go]: use `path.Join()` for windows support (this codebase does have windows support, it appears)

File: segment-kube.go (hunk 1 above)
Comment (justjanne) [reply segment-kube.go]: Maybe we should make this a config option?

**Comment by joemiller** [comment]
love it =)

File: segment-kube.go (hunk 2 above)
Comment (justjanne) [reply segment-kube.go]: Yes, surprisingly, it does have Windows support – and apparently, it actually works on Windows.

File: segment-kube.go (hunk 1 above)
Comment (joemiller) [reply segment-kube.go]: +1

File: segment-kube.go (hunk 2 above)
Comment (justjanne) [reply segment-kube.go]: I agree, this should be changed.

**Comment by spheromak** [comment]
@justjanne @joemiller cleaned up the README, and add comment/todo on the gke naming.

**Comment by justjanne** [comment]
There are still issues with this PR — would it be okay for you if I reimplement this myself?

**Comment by spheromak** [comment]
@justjanne sure do whatever you want :) I already use it everyday

**Comment by spheromak** [comment]
@justjanne sorry I realized my vim still stripped newlines, after I thought I had told it not to. Should have the README mods to the minimal stuff now.

**Comment by justjanne** [comment]
This change is included in v1.8.0

**Pull request closed without merging**